src/hooks/permissionedPools/PermissionedPositionManager.sol

```diff
@@ -36,17 +37,25 @@ contract PermissionedPositionManager is PositionManager {
         PERMISSIONED_HOOKS = _permissionedHooks;
     }
 
-    /// @dev Disables transfers of the ERC721 liquidity position tokens
-    function transferFrom(address, address, uint256) public pure override {
-        revert("Transfer disabled");
+    /// @dev Only allow admins of permissioned tokens to transfer positions that contain their tokens
+    function transferFrom(address from, address to, uint256 id) public override onlyIfPoolManagerLocked {
+        (PoolKey memory poolKey,) = getPoolAndPositionInfo(id);
+        address admin1 = _getOwner(poolKey.currency0);
+        address admin2 = _getOwner(poolKey.currency1);
+        if (msg.sender != admin1 && msg.sender != admin2) {
+            revert Unauthorized();
```

Comment:
what if both are permissioned and msg.sender is only admin on one? it seems we should have a check where they need to be admin of each one independently iff they are permissioned?

Reply:
this makes sense to me! thanks for outlining those cases